src/plugins/CouchDBSearchFolder/plugin.js

```diff
@@ -35,9 +35,13 @@ export default function (folderName, couchPlugin, searchFilter) {
         );
       },
       load() {
-        return couchProvider.getObjectsByFilter(searchFilter).then((objects) => {
-          return objects.map((object) => object.identifier);
-        });
+        if (searchFilter.viewName !== undefined) {
```

Comment:
You mentioned that you create the code to be more understandable. I think that you can use an `if(cond) return couchProvider` instead of the if else, if you get me?

Reply:
Multiple return statements is definitely considered an anti-pattern by some. let me see what I can do. Thanks!